### PR TITLE
Revamp rich text components folder structure

### PR DESCRIPTION
**Pull request by vivinkrishna-ni**
# Pull Request

## 🤨 Rationale

Reorganize the folder structure of the `nimble-rich-text-editor` and `nimble-rich-text-viewer` to share the common functionalities between components.

Part of https://github.com/ni/nimble/issues/1447

## 👩‍💻 Implementation

Revamped folder structure in nimble-components:
```
/rich-text
    /editor
        editor component files...
    /viewer
        viewer component files...
    /specs
        README.md (common spec file for both editor and viewer)
```

Revamped folder structure in nimble-angular:
```
/rich-text
    /editor
        editor component files...
    /viewer
        viewer component files...
```

Related discussion: https://github.com/ni/nimble/pull/1424#discussion_r1290998178

## 🧪 Testing

- Manually tested the components in the local storybook build.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/rich-text/viewer/index.ts
Comment (suseendran-ni) [inline packages/nimble-components/src/rich-text/viewer/index.ts:34]: Can we consider initializing this similar to `rich-text-editor` to maintain consistency?

**Review by suseendran-ni — APPROVED** [review]

File: packages/nimble-components/src/rich-text/models/markdown-parser.ts
Comment (vivinkrishna-ni) [inline packages/nimble-components/src/rich-text/models/markdown-parser.ts:1]: Reviewer's info:

This file is the new file just like the `markdown-serializer.ts`. However, the changes in the file are almost similar to the viewer's `index.ts` file containing the markdown parser logic, Git is comparing this file with the index file instead of showing it as a new file and showing the `index.ts` file as a new file.

File: packages/nimble-components/src/rich-text/viewer/index.ts
Comment (vivinkrishna-ni) [reply packages/nimble-components/src/rich-text/viewer/index.ts]: Yes updated as suggested.

File: packages/nimble-components/src/rich-text/models/markdown-serializer.ts
Comment (m-akinc) [inline packages/nimble-components/src/rich-text/models/markdown-serializer.ts:16]:
```suggestion
    private readonly markdownSerializer = this.initializeMarkdownSerializerForTipTap();
```

File: angular-workspace/projects/ni/nimble-angular/rich-text-editor/nimble-rich-text-editor.directive.ts
Comment (m-akinc) [inline angular-workspace/projects/ni/nimble-angular/rich-text-editor/nimble-rich-text-editor.directive.ts:1]: I believe we want the folder structure here to match nimble-components, i.e. `rich-text/editor` and `rich-text/viewer`. Is that right @rajsite?

**Review by m-akinc — APPROVED** [review]

File: angular-workspace/projects/ni/nimble-angular/rich-text-editor/nimble-rich-text-editor.directive.ts
Comment (rajsite) [reply angular-workspace/projects/ni/nimble-angular/rich-text-editor/nimble-rich-text-editor.directive.ts:1]: Absolutely!

Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text-editor/nimble-rich-text-editor.directive.ts:1]: Updated

File: packages/nimble-components/src/rich-text/viewer/specs/README.md
Comment (vikisekarNI) [inline packages/nimble-components/src/rich-text/viewer/specs/README.md:3]: nit: I think now we can avoid having two spec files, one for editor and another for viewer redirecting to editor. 

Instead can we have one specs folder under `rich-text` containing the actual spec HLD having both editor and viewer. If viewer branches out into different direction on the dependencies or logic (which is less unlikely), then we may need a separate spec with unique content for viewer, removing any redirection links to editor spec as given here.

But I don't have a strong preference on this suggestion. My concern was to only avoid this spec file which just redirects to another spec file. It's good to keep this open for Nimble code owner's advise on the preferred approach to maintain the spec.

**Review by vikisekarNI — APPROVED** [review]

File: packages/nimble-components/src/rich-text/models/markdown-parser.ts
Comment (jattasNI) [inline packages/nimble-components/src/rich-text/models/markdown-parser.ts:11]: Ideally we'd have unit test coverage for the markdown parser and serializer directly rather than relying on testing in the rich text editor and viewer specs. This might allow us to remove duplication across the component tests. Any thoughts on how feasible it would be to do that?

File: angular-workspace/projects/ni/nimble-angular/rich-text/editor/ng-package.json
Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/rich-text/editor/ng-package.json:1]: Can you make sure the schema path in the ng-package.json file is accurate? I'd expect a change in this file since it goes one layer deeper

File: angular-workspace/projects/ni/nimble-angular/rich-text/editor/testing/ng-package.json
Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/rich-text/editor/testing/ng-package.json:1]: Can you make sure the schema path in the ng-package.json file is accurate? I'd expect a change in this file since it goes one layer deeper

File: angular-workspace/projects/ni/nimble-angular/rich-text/viewer/ng-package.json
Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/rich-text/viewer/ng-package.json:1]: Can you make sure the schema path in the ng-package.json file is accurate? I'd expect a change in this file since it goes one layer deeper

File: angular-workspace/projects/ni/nimble-angular/rich-text/viewer/testing/ng-package.json
Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/rich-text/viewer/testing/ng-package.json:1]: Can you make sure the schema path in the ng-package.json file is accurate? I'd expect a change in this file since it goes one layer deeper

File: angular-workspace/projects/ni/nimble-angular/rich-text/editor/ng-package.json
Comment (rajsite) [reply angular-workspace/projects/ni/nimble-angular/rich-text/editor/ng-package.json:1]: @mollykreis do you know what would validate these paths are correct? Is it the Angular vscode plugin or something?

File: packages/nimble-components/src/rich-text/editor/index.ts
Comment (rajsite) [inline packages/nimble-components/src/rich-text/editor/index.ts:150]: Can these be static and shared across all component instances? If so that might be a change to make in a follow-up PR. This PR is probably already too large a scope changing folder hierarchy and refactoring implementation at the same-time so lets not grow it further.

File: packages/nimble-components/src/rich-text/models/markdown-parser.ts
Comment (rajsite) [reply packages/nimble-components/src/rich-text/models/markdown-parser.ts:11]: That sounds like a good idea. And I think I'd advocate not doing the folder structure change and the sharing of the model logic in the same PR. I think we should split the PRs so this change is just the folder structure re-organization and we can review de-duplicating the logic and tests in a follow-on PR.

File: packages/nimble-components/src/rich-text/models/markdown-parser.ts
Comment (vivinkrishna-ni) [reply packages/nimble-components/src/rich-text/models/markdown-parser.ts]: As we are moving the model file logic to a different PR, we can ignore the above-mentioned info.

File: angular-workspace/projects/ni/nimble-angular/rich-text/editor/ng-package.json
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text/editor/ng-package.json:1]: It should be one layer deeper as you mentioned. I have updated the same here.

File: angular-workspace/projects/ni/nimble-angular/rich-text/editor/testing/ng-package.json
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text/editor/testing/ng-package.json:1]: It should be one layer deeper as you mentioned. I have updated the same here.

File: angular-workspace/projects/ni/nimble-angular/rich-text/viewer/ng-package.json
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text/viewer/ng-package.json:1]: It should be one layer deeper as you mentioned. I have updated the same here.

File: angular-workspace/projects/ni/nimble-angular/rich-text/viewer/testing/ng-package.json
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text/viewer/testing/ng-package.json:1]: It should be one layer deeper as you mentioned. I have updated the same here.

File: packages/nimble-components/src/rich-text/models/markdown-parser.ts
Comment (vivinkrishna-ni) [reply packages/nimble-components/src/rich-text/models/markdown-parser.ts]: Thanks for the feedback. We will address this in the follow-up PR.

File: packages/nimble-components/src/rich-text/editor/index.ts
Comment (vivinkrishna-ni) [reply packages/nimble-components/src/rich-text/editor/index.ts]: Yes, It can be static and we will address it in the follow-up PR.

**Review by jattasNI — APPROVED** [review]

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/src/rich-text/viewer/specs/README.md
Comment (rajsite) [reply packages/nimble-components/src/rich-text/viewer/specs/README.md]: Looks good! We've also made separate documents for large new features and linked them from the main spec which can be another approach if large features are added in the future.

**Comment by rajsite** [comment]
@ vivinkrishna-ni looks like you have a merge conflict to resolve, once that is done feel free to merge. Don't need to wait on @atmgrifter00 or @mollykreis

**Pull request merged**
